### PR TITLE
Adding a workflow to create AzDO bugs from 'Issues' created on the repo

### PR DESCRIPTION
**Pull request by harsha-bhushan**
### What does this Pull Request accomplish?

This PR enables creation of bugs on AzDO (in the appropriate area path) when an issue on the grpc-device github repo is created.
A workflow was created using the steps outlined in this video: https://www.youtube.com/watch?v=I9ItftRT9dU
Using the yml script provided here: https://github.com/marketplace/actions/github-issues-to-azure-devops

(note: the other way - syncing any changes on AzDO back to github will not be done.)

### Why should this Pull Request be merged?

This avoids the below manual efforts:
1. Tracking issues opened on the grpc-device repo
2. Creating corresponding AzDO bugs so that they show up on domain dashboard & can be discussed during standups
3. Closing bugs after fixing the issue

### What testing has been done?
Unable to test this on the branch. Once merged, need to manually test the below scenarios:
Creating an  issue on github creates a bug in AzDO in the right project area path
Closing an issue on github closes the bug in AzDO

### CODE REVIEW

File: .github/workflows/sync_github_issues_to_azdo.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    types:
+      [opened, edited, deleted, closed, reopened, labeled, unlabeled, assigned]
+
+jobs:
+  alert:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: danhellem/github-actions-issue-to-work-item@master
+        env:
+          ado_token: "${{ secrets.ADO_PERSONAL_ACCESS_TOKEN }}"
```
Comment (harsha-bhushan) [inline .github/workflows/sync_github_issues_to_azdo.yml:14]: note: I have created a personal access token & added it under secrets

Comment (SparkingSpork) [reply .github/workflows/sync_github_issues_to_azdo.yml:14]: I think it's preferable to get tokens created for this by the organization/project owners in GitHub and AzDO respectively. If we use a personal token you created, we have to rely on you in the future for maintaining it and keeping it from expiring or swap to a new token owned by someone else occasionally.

I know for some other workflows @zhindes requested an organizational level secret (such as [here](https://github.com/ni/grpc-device/blob/main/.github/workflows/build_cmake.yml#L228)). We should at least entertain that approach instead, which will result in an organizational secret available on all NI repositories: 
image

Comment (SparkingSpork) [reply .github/workflows/sync_github_issues_to_azdo.yml:14]: If you'd like to get this in to test it before getting the new access tokens created, I'm OK with that as well.

**Review by SparkingSpork — APPROVED** [review]

File: .github/workflows/sync_github_issues_to_azdo.yml (hunk 1 above)
Comment (zhindes) [reply .github/workflows/sync_github_issues_to_azdo.yml:14]: Yeah, I would recommend an org-wide secret that isn't a personal token. Ping one of the NI AzDO/GitHub admins to get that created and setup. I usually talk directly to David Corrigan for stuff like this, but he may direct you to a better method of reaching out.

**Review by astarche — APPROVED** [review]

**Review by doshirohan — APPROVED** [review]

File: .github/workflows/sync_github_issues_to_azdo.yml (hunk 1 above)
Comment (davidcorrigan714) [reply .github/workflows/sync_github_issues_to_azdo.yml:14]: Sorry on the delay on this, had a busy week last week and want to dig into the security features of GitHub a bit before setting up a token with R/W access to AzDO work items. I'll try and do that in the next few days.

Comment (davidcorrigan714) [reply .github/workflows/sync_github_issues_to_azdo.yml]: Added the tokens. The AzDO one is scoped pretty narrowly to just that area path so if that changes we'll need to update it. That area path should really be an ID and not a string too but that's not something you all could fix - it's just not as nice if area paths get rearranged in DevCentral.

**Pull request merged**